tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -349,6 +354,16 @@ public bool DiagnosticSourceEnabled
         /// </summary>
         public bool StartupDiagnosticLogEnabled { get; set; }
 
+        /// <summary>
+        /// Gets or sets the maximum length of an outgoing propagation header's value ("x-datadog-tags")
+        /// when injecting it into downstream service calls.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.TagPropagation.HeaderMaxLength"/>
+        /// <remarks>
+        /// This value is not used when extracting an incoming propagation header from an upstream service.
+        /// </remarks>
+        public int PropagationHeaderMaxLength { get; set; }
```

Comment:
Don't want we users to be able to configure tracer settings in code?

Reply:
IMO, no 😉 It causes us nothing but headaches, so the more we can push people away from it the better IMO. We've (I've) been favouring not adding _new_ features to `TracerSettings` generally, but I don't know that it's an explicit decision we've made...

Reply:
> I don't know, they have to set global env vars in that case

Yes obv. But as we tell them to add a package, they could start thinking more about configuring in code.  Plus it would mean, main setup is for SRE, the rest for them. But one could argue they can still use the app config files

Reply:
> Don't forget the custom instrumentation nuget, too

I try to, whenever possible 😛 

> That's how I would do it if I was using the nuget :)

I guess that's where we differ 😉

Reply:
Changed to `internal`. We can make it public later if needed.